### PR TITLE
[Enhancement] put `io_task_per_scan_operator` to session variable

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Right now this configuration is a static BE conf, which is bad for tuning. 

And we have found the case that, high value of this item is good for accesing remote storage when latency is high, but no ood(and not bad) when latency is ok. However, higher value means more memory consumption, which will incur OOM. 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by dirtysalt** [comment]
Fix a warning

```
In file included from /home/disk4/zhangyan/StarRocks/be/src/exprs/agg/factory/aggregate_factory.hpp:25,
                 from /home/disk4/zhangyan/StarRocks/be/src/exprs/agg/factory/aggregate_resolver.hpp:23,
                 from /home/disk4/zhangyan/StarRocks/be/src/storage/column_aggregate_func.cpp:20:
In member function ‘void starrocks::ArrayAggAggregateState::update(starrocks::FunctionContext*, const starrocks::Column&, size_t, size_t, size_t)’,
    inlined from ‘virtual void starrocks::ArrayAggAggregateFunction::merge(starrocks::FunctionContext*, const starrocks::Column*, starrocks::AggDataPtr, size_t) const’ at /home/disk4/zhangyan/StarRocks/be/src/exprs/agg/array_agg.h:134:37,
    inlined from ‘void starrocks::AggregateFunctionBatchHelper<State, Derived>::merge_batch_single_state(starrocks::FunctionContext*, starrocks::AggDataPtr, const starrocks::Column*, size_t, size_t) const [with State = starrocks::ArrayAggAggregateState; Derived = starrocks::ArrayAggAggregateFunction]’ at /home/disk4/zhangyan/StarRocks/be/src/exprs/agg/aggregate.h:375:53,
    inlined from ‘virtual void starrocks::AggFuncBasedValueAggregator::aggregate_batch_impl(int, int, const ColumnPtr&)’ at /home/disk4/zhangyan/StarRocks/be/src/storage/column_aggregate_func.cpp:289:44:
/home/disk4/zhangyan/StarRocks/be/src/exprs/agg/array_agg.h:36:27: warning: ‘this’ pointer is null [-Wnonnull]
   36 |             ctx->set_error(s.c_str(), false);
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
In file included from /home/disk4/zhangyan/StarRocks/be/src/exprs/expr_context.h:42,
                 from /home/disk4/zhangyan/StarRocks/be/src/runtime/chunk_cursor.h:21,
                 from /home/disk4/zhangyan/StarRocks/be/src/exec/sorting/sorting.h:22,
                 from /home/disk4/zhangyan/StarRocks/be/src/exprs/agg/array_agg.h:21,
                 from /home/disk4/zhangyan/StarRocks/be/src/exprs/agg/factory/aggregate_factory.hpp:25,
                 from /home/disk4/zhangyan/StarRocks/be/src/exprs/agg/factory/aggregate_resolver.hpp:23,
                 from /home/disk4/zhangyan/StarRocks/be/src/storage/column_aggregate_func.cpp:20:
/home/disk4/zhangyan/StarRocks/be/src/exprs/function_context.h: In member function ‘virtual void starrocks::AggFuncBasedValueAggregator::aggregate_batch_impl(int, int, const ColumnPtr&)’:
/home/disk4/zhangyan/StarRocks/be/src/exprs/function_context.h:90:10: note: in a call to non-static member function ‘void starrocks::FunctionContext::set_error(const char*, bool)’
   90 |     void set_error(const char* error_msg, const bool is_udf = true);
      |          ^~~~~~~~~
```

**Comment by dirtysalt** [comment]
Query/Time,Mem | SR(dop=52,IO=16) | SR(dop=24,io=16) | SR(dop=32,io=4) | SR(dop=52, IO=4) | Presto
-- | -- | -- | -- | -- | --
Q01 | 5322ms / 18.27GB | 5643ms /17.54GB | 4847ms / 7.35GB | 5138 / 12.19GB | 32961ms / 8.09GB
Q03 | 4137ms / 14.50GB | 4674ms / 11.75GB | 3673ms / 6.10GB | 4570 / 10.20GB | 76034ms / 6.98GB
Q04 | 2251ms / 5.86GB | 1589ms / 5.17GB | 2034ms / 2.45GB | 1738 / 3.76GB | 39353ms / 3.68GB
Q06 | 3846ms / 11.27GB | 3047ms / 11.82GB | 3286ms / 4.93GB | 3447 / 7.57GB | 42161ms / 1.64KB
Q07 | 6118ms / 14.65GB | 4930ms / 15.2GB | 9123ms / 6.48GB | 5386 / 9.54GB | 127002ms / 7.93GB
Q09 | 8023ms / 30.52GB | 7403ms / 24.47GB | 8056ms / 16.23GB | 9502 / 21.79GB | 61800ms / 18.9GB
Q10 | 4913ms / 14.04GB | 4929ms / 12.66GB | 5783ms / 7.83GB | 5473 / 9.27GB | 69752ms / 6.30GB

File: be/src/io/shared_buffered_input_stream.h [hunk 1]
```diff
@@ -76,8 +75,10 @@ class SharedBufferedInputStream : public SeekableInputStream {
         int64_t ref_count;
         std::vector<uint8_t> buffer;
     };
+    Status _get_bytes(SharedBuffer& sb, const uint8_t** buffer, size_t offset, size_t* nbytes);
     StatusOr<SharedBuffer*> _find_shared_buffer(size_t offset, size_t count);
     std::shared_ptr<SeekableInputStream> _stream;
+    std::string _filename;
```
Comment (DorianZheng) [inline be/src/io/shared_buffered_input_stream.h:81]: I don't find any code using this variable

File: be/src/exprs/agg/array_agg.h [hunk 2]
```diff
@@ -32,20 +32,20 @@ namespace starrocks {
 struct ArrayAggAggregateState {
     void update(FunctionContext* ctx, const Column& column, size_t index, size_t offset, size_t count) {
         if (UNLIKELY(index >= data_columns->size())) {
-            ctx->set_error(
-                    (std::string(fmt::format("index {} is larger than column size {}.", index, data_columns->size())))
-                            .c_str(),
-                    false);
+            if (ctx != nullptr) {
```
Comment (DorianZheng) [inline be/src/exprs/agg/array_agg.h:35]: why change this in this PR

File: be/src/io/shared_buffered_input_stream.h (hunk 1 above)
Comment (dirtysalt) [reply be/src/io/shared_buffered_input_stream.h:81]: No. I used this variable before, but I forget to clean. 

This variable perhaps will be used in future. 

I can remove them if you mind.

**Review by DorianZheng — COMMENTED** [review]
left comments, rest LGTM

File: be/src/exprs/agg/array_agg.h (hunk 2 above)
Comment (dirtysalt) [reply be/src/exprs/agg/array_agg.h:35]: Because of compiler warning, I already commented.

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**